From ba0b41445ee8a911ed780564788fbd58c34a62fa Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 21 Nov 2022 14:18:57 +0000
Subject: [PATCH] overlays: audremap: Include the fsels values

Add the alt function selectors, and document that pins_40_45 is not
suitable for BCM2711 because the pins are split across the two PWM
interfaces.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README               |  3 ++-
 arch/arm/boot/dts/overlays/audremap-overlay.dts | 12 ++++++++----
 2 files changed, 10 insertions(+), 5 deletions(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -685,7 +685,8 @@ Params: swap_lr                 Reverse
         pins_12_13              Select GPIOs 12 & 13 (default)
         pins_18_19              Select GPIOs 18 & 19
         pins_40_41              Select GPIOs 40 & 41
-        pins_40_45              Select GPIOs 40 & 45
+        pins_40_45              Select GPIOs 40 & 45 (don't use on BCM2711 - the
+                                pins are on different controllers)
 
 
 Name:   balena-fin
--- a/arch/arm/boot/dts/overlays/audremap-overlay.dts
+++ b/arch/arm/boot/dts/overlays/audremap-overlay.dts
@@ -22,12 +22,16 @@
 		swap_lr = <&frag0>, "swap_lr?";
 		enable_jack = <&frag0>, "enable_jack?";
 		pins_12_13 = <&frag0>,"brcm,pins:0=12",
-		             <&frag0>,"brcm,pins:4=13";
+		             <&frag0>,"brcm,pins:4=13",
+			     <&frag0>,"brcm,function:0=4";
 		pins_18_19 = <&frag0>,"brcm,pins:0=18",
-		             <&frag0>,"brcm,pins:4=19";
+		             <&frag0>,"brcm,pins:4=19",
+			     <&frag0>,"brcm,function:0=2";
 		pins_40_41 = <&frag0>,"brcm,pins:0=40",
-		             <&frag0>,"brcm,pins:4=41";
+		             <&frag0>,"brcm,pins:4=41",
+			     <&frag0>,"brcm,function:0=4";
 		pins_40_45 = <&frag0>,"brcm,pins:0=40",
-		             <&frag0>,"brcm,pins:4=45";
+		             <&frag0>,"brcm,pins:4=45",
+			     <&frag0>,"brcm,function:0=4";
 	};
 };
